### PR TITLE
Modeler 2.4c portfield substitute

### PR DESCRIPTION
**Pull request by payetvin**

### CODE REVIEW

**Review by flomnes — COMMENTED** [review]
Please derive a new class from `CloneVisitor`.

File: src/solver/expressions/visitors/PortfieldSubstitutionVisitor.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+PortfieldSubstitutionVisitor::PortfieldSubstitutionVisitor(Registry<Nodes::Node>& registry, SubstitutionContext& ctx):
+    CloneVisitor(registry),
+    ctx_(ctx),
+    registry_(registry)
+{
+}
+
+Nodes::Node* PortfieldSubstitutionVisitor::visit(const Nodes::PortFieldNode* port_field_node)
+{
+    return registry_.create<Nodes::PortFieldNode>(port_field_node->getPortName(),
```
Comment (a-zakir) [inline src/solver/expressions/visitors/PortfieldSubstitutionVisitor.cpp]: here you can copy-paste SubstitutionVisitor::visit

File: src/solver/expressions/include/antares/solver/expressions/visitors/CloneVisitor.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ class CloneVisitor: public NodeVisitor<Nodes::Node*>
      *
      * @param registry The registry used for creating new nodes.
      */
-    explicit CloneVisitor(Registry<Nodes::Node>& registry);
+    CloneVisitor(Registry<Nodes::Node>& registry);
```
Comment (flomnes) [inline src/solver/expressions/include/antares/solver/expressions/visitors/CloneVisitor.h]: Please revert this

File: src/solver/expressions/include/antares/solver/expressions/visitors/PortfieldSubstitutionVisitor.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * @brief Represents a visitor for substituting portfield nodes in a syntax tree.
+ */
+class PortfieldSubstitutionVisitor: public CloneVisitor
+{
+public:
+    PortfieldSubstitutionVisitor(Registry<Nodes::Node>& registry,
+                                 PortfieldSubstitutionContext& ctx);
+
+    PortfieldSubstitutionContext& ctx_;
+    Registry<Nodes::Node>& registry_;
```
Comment (flomnes) [inline src/solver/expressions/include/antares/solver/expressions/visitors/PortfieldSubstitutionVisitor.h]: unused

File: src/solver/expressions/visitors/EvaluationContext.cpp [hunk 4]
```diff
@@ -18,4 +18,10 @@ double EvaluationContext::getParameterValue(const std::string& key) const
 {
     return parameters_.at(key);
 }
+
+Nodes::Node* EvaluationContext::getNodeToSubstitute(Nodes::PortFieldNode* key) const
+{
+    return nodesToSubstitutes_.at(key);
+}
+
```
Comment (flomnes) [inline src/solver/expressions/visitors/EvaluationContext.cpp]: unused

File: src/solver/expressions/include/antares/solver/expressions/visitors/EvaluationContext.h [hunk 5]
```diff
@@ -46,6 +48,8 @@ class EvaluationContext
      */
     double getParameterValue(const std::string& key) const;
 
+    Nodes::Node* getNodeToSubstitute(Nodes::PortFieldNode*) const;
+
```
Comment (flomnes) [inline src/solver/expressions/include/antares/solver/expressions/visitors/EvaluationContext.h]: unused

File: src/solver/expressions/include/antares/solver/expressions/visitors/EvaluationContext.h [hunk 6]
```diff
@@ -56,6 +60,11 @@ class EvaluationContext
      * @brief A map storing variable values.
      */
     std::map<std::string, double> variables_;
+
+    /**
+     * @brief A map storing nodes to substitute with portfield.
+     */
+    std::map<Nodes::PortFieldNode*, Nodes::Node*> nodesToSubstitutes_;
```
Comment (flomnes) [inline src/solver/expressions/include/antares/solver/expressions/visitors/EvaluationContext.h]: unused

**Pull request closed without merging**